File: kraken/src/main/java/glitch/kraken/object/json/list/ChatRooms.java
package glitch.kraken.object.json.list;

import com.google.common.collect.ImmutableList;
import com.google.gson.annotations.SerializedName;
import glitch.api.objects.json.interfaces.OrdinalList;
import glitch.kraken.object.json.ChatRoom;
import lombok.AccessLevel;
import lombok.Data;
import lombok.Getter;

@Data
public class ChatRooms implements OrdinalList<ChatRoom> {
    private final ImmutableList<ChatRoom> data;
    @SerializedName("_total")
    @Getter(AccessLevel.NONE)
    private final int total;

    @Override
    public int size() {
        return total;
    }
}

File: kraken/src/main/kotlin/glitch/kraken/object/json/collections/ChatRooms.kt
package glitch.kraken.`object`.json.collections

import com.google.gson.annotations.SerializedName
import glitch.api.objects.json.interfaces.OrdinalList
import glitch.kraken.`object`.json.ChatRoom

data class ChatRooms(
        override val data: List<ChatRoom>,
        @SerializedName("_total")
        override val size: Int
) : OrdinalList<ChatRoom>
